File: 프로그래머스/lv2/12909. 올바른 괄호/올바른 괄호.java
import java.util.LinkedList;
import java.util.Queue;
import java.lang.String;
import java.util.Stack;

class Solution {
    boolean solution(String s) {
        boolean answer = true;
        Stack<Character> stack=new Stack<>();
        if(s.charAt(0)==(')')){
            return false;
        }
        for(int i=0;i<s.length();i++){
            if(!stack.isEmpty()&&s.charAt(i)==(')')){
                stack.pop();
            } else stack.push(s.charAt(i));
        }
        if(!stack.isEmpty()){return false;}

        return true;
    }
}

File: 프로그래머스/lv2/12909. 올바른 괄호/올바른 괄호.py
def solution(s):
    stack = []
    for i in s:
        if i == '(':
            stack.append(i)
        else:
            if stack == []: # )이  나왔는데 스택이 비어있는경우
                return False
            else:
                stack.pop()
    if stack != []: # 다 끝났는데 스택이 비어있지않다면
        return False
    return True
